controllers/cluster_controller.go

```diff
@@ -640,31 +642,60 @@ func checkCoordinatorValidity(cluster *fdbtypes.FoundationDBCluster, status *fdb
 			continue
 		}
 
-		var address string
+		var ipAddress fdbtypes.ProcessAddress
 		for _, addr := range addresses {
 			if addr.Flags["tls"] == cluster.Spec.MainContainer.EnableTLS {
-				address = addr.String()
+				ipAddress = addr
 				break
 			}
 		}
 
-		_, isCoordinator := coordinatorStatus[address]
-		if isCoordinator && !process.Excluded && !pendingRemoval {
-			coordinatorStatus[address] = true
+		coordinatorAddress := ""
+		_, isCoordinatorWithIP := coordinatorStatus[ipAddress.String()]
+		if isCoordinatorWithIP {
+			coordinatorAddress = ipAddress.String()
 		}
 
-		if isCoordinator {
+		dnsName := process.Locality["dns_name"]
+		dnsAddress := fdbtypes.ProcessAddress{
+			StringAddress: dnsName,
+			Port:          ipAddress.Port,
+			Flags:         ipAddress.Flags,
+		}
+		_, isCoordinatorWithDNS := coordinatorStatus[dnsAddress.String()]
+
+		if !isCoordinatorWithDNS {
+			dnsAddress = ipAddress
+			dnsAddress.FromHostname = true
+			_, isCoordinatorWithDNS = coordinatorStatus[dnsAddress.String()]
+		}
+
+		if isCoordinatorWithDNS {
+			coordinatorAddress = dnsAddress.String()
+		}
+
+		if coordinatorAddress != "" && !process.Excluded && !pendingRemoval {
+			coordinatorStatus[coordinatorAddress] = true
+		}
+
+		if coordinatorAddress != "" {
 			coordinatorZones[process.Locality[fdbtypes.FDBLocalityZoneIDKey]]++
 			coordinatorDCs[process.Locality[fdbtypes.FDBLocalityDCIDKey]]++
 
 			if !cluster.IsEligibleAsCandidate(process.ProcessClass) {
-				pLogger.Info("Process class of process is not eligible as coordinator", "class", process.ProcessClass, "address", address)
+				pLogger.Info("Process class of process is not eligible as coordinator", "class", process.ProcessClass, "address", ipAddress)
 				allEligible = false
 			}
+
+			useDNS := cluster.UseDNSInClusterFile() && dnsName != ""
+			if (isCoordinatorWithIP && useDNS) || (isCoordinatorWithDNS && !useDNS) {
+				pLogger.Info("Coordinator is not using the correct address", "coordinatorList", coordinatorStatus)
```

Comment:
```suggestion
				pLogger.Info("Coordinator is not using the correct address type", "coordinatorList", coordinatorStatus)
```

We might want to add thee expected address "type" to the log statement.